### PR TITLE
Introduce remote table entity

### PR DESCRIPTION
**Pull request by thomtrp**
We will require remote table entity to map distant table name and local foreign table name.
Introducing the entity:
- new source of truth to know if a table is sync or not
- created synchronously at the same time as metadata and foreign table

Adding a few more changes:
- exception rather than errors so the user can see these
- `pluralize` library that will allow to stop adding `Remote` suffix on names

### CODE REVIEW

File: packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts [hunk 1]
```diff
@@ -104,13 +106,76 @@ export class RemoteTableService {
...
+    }
+
+    const dataSourceMetatada =
+      await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        workspaceId,
+      );
+
+    const localTableName = getRemoteTableLocalName(input.name);
+
+    this.validateTableNameDoesNotExists(
```
Comment (Weiko) [inline packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts:135]: missing await here?

File: packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts [hunk 2]
```diff
@@ -104,13 +106,76 @@ export class RemoteTableService {
       throw new NotFoundException('Remote server does not exist');
     }
 
-    const remoteTable = await this.createForeignTableAndMetadata(
+    if (!input.schema) {
```
Comment (Weiko) [inline packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts:109]: Let's save one query by checking this earlier! 😄

File: packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts [hunk 3]
```diff
@@ -104,13 +106,76 @@ export class RemoteTableService {
...
+
+    const localTableName = getRemoteTableLocalName(input.name);
+
+    this.validateTableNameDoesNotExists(
+      localTableName,
+      workspaceId,
+      dataSourceMetatada.schema,
+    );
+
+    const remoteTableEntity = this.remoteTableRepository.create({
```
Comment (Weiko) [inline packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts:141]: same here?

Comment (thomtrp) [reply packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts:141]: this is not an async function this one :)

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**